prime-router/src/main/kotlin/fhirengine/engine/FHIRReceiver.kt

```diff
@@ -97,94 +91,31 @@ class FHIRReceiver(
         actionLogger: ActionLogger,
         actionHistory: ActionHistory,
     ): List<FHIREngineRunResult> {
-        val contextMap = createLoggingContextMap(queueMessage, actionHistory)
         // Use the logging context for tracing
-        withLoggingContext(contextMap) {
-            actionLogger.setReportId(queueMessage.reportId)
-            val sender = getSender(queueMessage, actionLogger, actionHistory) ?: return emptyList()
-
-            // Process the message if no errors occurred
-            return handleSuccessfulProcessing(queueMessage, sender, actionLogger, actionHistory)
-        }
-    }
-
-    /**
-     * Creates the logging context map.
-     *
-     * @param queueMessage The queue message containing details about the report.
-     * @param actionHistory The action history related to receiving the report.
-     * @return The logging context map.
-     */
-    private fun createLoggingContextMap(
-        queueMessage: FhirReceiveQueueMessage,
-        actionHistory: ActionHistory,
-    ): Map<MDCUtils.MDCProperty, Any> = mapOf(
+        withLoggingContext(
+            mapOf(
             MDCUtils.MDCProperty.ACTION_NAME to actionHistory.action.actionName.name,
             MDCUtils.MDCProperty.REPORT_ID to queueMessage.reportId,
             MDCUtils.MDCProperty.BLOB_URL to queueMessage.blobURL,
         )
-
-    /**
-     * Retrieves the sender based on the queue message and logs any relevant errors.
-     *
-     * @param queueMessage The queue message containing details about the report.
-     * @param actionLogger The logger used to track actions and errors.
-     * @param actionHistory The action history related to receiving the report.
-     * @return The sender, or null if the sender was not found or is inactive.
-     */
-    private fun getSender(
-        queueMessage: FhirReceiveQueueMessage,
-        actionLogger: ActionLogger,
-        actionHistory: ActionHistory,
-    ): Sender? {
-        val clientId = queueMessage.headers[clientIdHeader]
-        val sender = clientId?.takeIf { it.isNotBlank() }?.let { settings.findSender(it) }
-
-        actionHistory.trackActionParams(queueMessage.headers.toString())
-
-        // Handle case where sender is not found
-        return if (sender == null) {
-            // Send an error event
-            reportEventService.sendSubmissionProcessingError(
```

Comment:
Rather than emit an event, we're now adding an item to the poison queue which will cause an alert and enables the engagement team to fix the configuration and requeue the message.